module/data/activity/attack-data.mjs

```diff
@@ -244,24 +245,38 @@ export default class AttackActivityData extends BaseActivityData {
   getDamageConfig(config={}) {
     const rollConfig = super.getDamageConfig(config);
 
-    // TODO: Handle ammunition
-    // If base part is present, add ammunition physical properties to it
-    // If ammo has its own damage, create a new damage part of it
-
-    // const ammo = this.hasAmmo ? this.actor.items.get(this.system.consume.target) : null;
-    // if ( ammo ) {
-    //   const properties = Array.from(ammo.system.properties).filter(p => CONFIG.DND5E.itemProperties[p]?.isPhysical);
-    //   if ( this.system.properties.has("mgc") && !properties.includes("mgc") ) properties.push("mgc");
-    //   const ammoConfigs = ammo.system.damage.parts.map((([formula, type]) => ({ parts: [formula], type, properties })));
-    //   if ( ammo.system.magicalBonus && ammo.system.magicAvailable ) {
-    //     rollConfigs[0].parts.push("@ammo");
-    //     properties.forEach(p => {
-    //       if ( !rollConfigs[0].properties.includes(p) ) rollConfigs[0].properties.push(p);
-    //     });
-    //     rollData.ammo = ammo.system.magicalBonus;
-    //   }
-    //   rollConfigs.push(...ammoConfigs);
-    // }
+    // Handle ammunition
+    const ammo = config.ammunition?.system;
+    if ( ammo ) {
+      const properties = Array.from(ammo.properties).filter(p => CONFIG.DND5E.itemProperties[p]?.isPhysical);
+      if ( this.item.system.properties?.has("mgc") && !properties.includes("mgc") ) properties.push("mgc");
+
+      // Add any new physical properties from the ammunition to the damage properties
+      for ( const roll of rollConfig.rolls ) {
+        for ( const property of properties ) {
+          if ( !roll.options.properties.includes(property) ) roll.options.properties.push(property);
+        }
+      }
+
+      // Add the ammunition's damage
+      if ( ammo.damage.base.formula ) {
+        const basePartIndex = rollConfig.rolls.findIndex(i => i.base);
+        const damage = ammo.damage.base.clone();
+        const rollData = this.getRollData();
+
+        // If mode is "replace" and base part is present, replace the base part
+        if ( ammo.damage.replace & (basePartIndex !== -1) ) {
+          damage.base = true;
+          rollConfig.rolls.splice(basePartIndex, 1, super._processDamagePart(damage, config, rollData));
```

Comment:
`this._processDamagePart`? Or are we deliberately skipping this layer of the implementation? It would mean that any subclasses that override `_processDamagePart` will not have their method called here.

```suggestion
          rollConfig.rolls.splice(basePartIndex, 1, this._processDamagePart(damage, config, rollData));
```